src/edu/washington/escience/myria/api/encoding/PerfEnforceTableEncoding.java

```diff
@@ -0,0 +1,55 @@
+/**
+ *
+ */
+package edu.washington.escience.myria.api.encoding;
+
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import edu.washington.escience.myria.RelationKey;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.io.AmazonS3Source;
+
+/**
+ * This class returns an encoding describing a table's schema.
+ */
+public class PerfEnforceTableEncoding {
+  public RelationKey relationKey;
+  public String type;
+  public AmazonS3Source source;
+  public Schema schema;
+  public Character delimiter;
+  public Set<Integer> keys;
+
+  public Set<Integer> corresponding_fact_key;
+
+  /**
+   * @param relationKey the name of the table
+   * @param type can be either a "fact" or "dimension" type
+   * @param source the location of the table's data in S3
+   * @param schema the schema of the table
+   * @param delimiter the delimiter used on the table's data in S3
+   * @param keys the column id of the primary key
+   * @param corresponding_fact_key the column id of the corresponding foreign key, primarily only used for dimension tables
+   */
+  @JsonCreator
+  public PerfEnforceTableEncoding(
+      @JsonProperty(value = "relationKey", required = true) final RelationKey relationKey,
```

Comment:
Not necessarily. I just didn't use the JSON encoding for ```PerfEnforceStatisticsEncoding```.  I will add it to keep it all consistent.